pageserver/src/tenant/timeline.rs

```diff
@@ -3964,7 +3967,11 @@ impl Timeline {
             (horizon_cutoff, pitr_cutoff, retain_lsns)
         };
 
-        let new_gc_cutoff = Lsn::min(horizon_cutoff, pitr_cutoff);
+        let mut new_gc_cutoff = Lsn::min(horizon_cutoff, pitr_cutoff);
+        let standby_flush_lsn = self.standby_flush_lsn.load();
+        if standby_flush_lsn != Lsn::INVALID && standby_flush_lsn < new_gc_cutoff {
```

Comment:
nit: it would be nice to avoid the `mut` and express the operation as a `min` rather than an `if <`

```
let new_gc_cutoff = if standby_flush_lsn != Lsn::INVALID {
  Lsn::min(standby_flush_lsn, new_gc_cutoff)
} else {
  new_gc_cutoff
}
```

Reply:
Done